wal.go

```diff
@@ -389,10 +389,10 @@ func (wal *WriteAheadLog) findStartingLogFile(offset int64, files []string) (i i
 		if err != nil {
 			return -1, -1, err
 		}
-		if previousOffset <= offset && offset <= startingOffset {
+
+		if offset <= startingOffset {
```

Comment:
The format for the file name is wal.{file_Count}.{starting_offset_in_wal_file}
Here `starting_offset_in_wal_file` tells us what offset the file starts from. We later increment this in the `func (wal *WriteAheadLog) Replay`. This offset helps us to maintain the track of the offset so far. 

With the changes you have proposed, we are not updating this offset.